public/docs/_examples/reactive-forms/ts/app/hero-detail-versions.component.ts

```diff
@@ -0,0 +1,264 @@
+import { FormArray, FormBuilder, FormControl, FormGroup } from '@angular/forms';
+import { Address, Hero, states } from './data-model';
+
+//////// 1 ////////////////////////
+
+@Component({
+  moduleId: module.id,
+  selector: 'hero-detail-1',
+  templateUrl: './hero-detail-1.component.html'
+})
+// #docregion v1
+export class HeroDetailComponent1 {
+  heroForm = new FormGroup ({
+    name: new FormControl()
+  });
+}
+// #enddocregion v1
+
+//////// 2 ////////////////////////
+
+@Component({
+  moduleId: module.id,
+  selector: 'hero-detail-2',
+  templateUrl: './hero-detail-2.component.html'
+})
+// #docregion v2
+export class HeroDetailComponent2 {
+  heroForm: FormGroup; // <--- heroForm is of type FormGroup
+
+  constructor(private fb: FormBuilder) { // <--- inject FormBuilder
+    this.createForm();
+  }
+
+  createForm() {
+    this.heroForm = this.fb.group({
+      name: '', // <--- the FormControl called "name"
+    });
+  }
+}
+// #enddocregion v2
+
+//////// 3 ////////////////////////
+
+@Component({
+  moduleId: module.id,
+  selector: 'hero-detail-3',
+  templateUrl: './hero-detail-3.component.html'
+})
+// #docregion v3
+export class HeroDetailComponent3 {
+  heroForm: FormGroup;
+  states = states;
+
+  constructor(private fb: FormBuilder) {
+    this.createForm();
+  }
+
+  createForm() {
+    this.heroForm = this.fb.group({
+      name: '',
+      street: '',
+      city: '',
+      state: '',
+      zip: ''
+    });
+  }
+}
+// #enddocregion v3
+
+////////// 4 /////////////////////
+
+@Component({
+  moduleId: module.id,
+  selector: 'hero-detail-4',
+  templateUrl: './hero-detail-4.component.html'
+})
+// #docregion v4
+export class HeroDetailComponent4 {
+  heroForm: FormGroup;
+  states = states;
+
+  constructor(private fb: FormBuilder) {
+    this.createForm();
+  }
+
+  createForm() {
+    this.heroForm = this.fb.group({ // <-- the parent FormGroup
+      name: '',
+      address: this.fb.group({ // <-- the child FormGroup
+        street: '',
+        city: '',
+        state: '',
+        zip: ''
+      })
+    });
+  }
+}
+// #enddocregion v4
+
+////////// 5 ////////////////////
+
+@Component({
+  moduleId: module.id,
+  selector: 'hero-detail-5',
+  templateUrl: './hero-detail-4.component.html'
+})
+// #docregion v5
+export class HeroDetailComponent5 implements OnChanges {
+  @Input() hero: Hero;
+
+  heroForm: FormGroup;
+  states = states;
+
+  constructor(private fb: FormBuilder) {
+    this.createForm();
+  }
+
+  createForm() {
+    // #docregion hero-form-model
+    this.heroForm = this.fb.group({
+      name: '',
+      address: this.fb.group({
+        street: '',
+        city: '',
+        state: '',
+        zip: ''
+      })
+    });
+    // #enddocregion hero-form-model
+  }
+
+  // #docregion patch-value-on-changes
+  ngOnChanges() { // <-- wrap patchValue in ngOnChanges
+    this.heroForm.reset();
+    // #docregion patch-value
+    this.heroForm.patchValue({
```

Comment:
Because (a) this section is about `patchValue` and (b) didn't know you could do that.

Even though you can, does it make sense when we're trying to focus on `patchValue`?